src/dune_file.ml

```diff
@@ -1842,18 +1849,22 @@ module Coq = struct
        and+ flags = field_oslu "flags"
        and+ modules = modules_field "modules"
        and+ libraries = field "libraries" Lib_deps.decode ~default:[]
+       and+ theories = field "theories" Lib_deps.decode ~default:[]
        and+ enabled_if = enabled_if
        in
        let name =
          let (loc, res) = name in
-         (loc, Lib_name.Local.validate (loc, res) ~wrapped:None)
+         match res with
+         | Ok t | Warn t -> (loc, t)
```

Comment:
You should fail on `Warn` here as well. Dune does not fail because of backwards compatibility issues. There's no reason for the coq plugin to inherit that.

By the way, this is why I think it's a good reason to avoid reusing `Lib_name` at all on the coq side. You should really rely on your own validation.

Reply:
Indeed this is pending; the time we stop piggybacking on `Lib_name` has come; at least in this version of the language, we need to be able to name Coq libs such that they share some common root such as `(name foo.bar)` `(name foo.baz)`.